runtime/compiler/env/DependencyTable.hpp

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#ifndef DEPENDENCYTABLE_INCL
+#define DEPENDENCYTABLE_INCL
+
+#include "env/J9PersistentInfo.hpp"
+#include "env/PersistentCollections.hpp"
+#include "env/VMJ9.h"
+
+class TR_J9SharedCache;
+
+struct OffsetEntry
+   {
+   PersistentUnorderedSet<J9Class *> _loadedClasses;
```

Comment:
Will this work on all build compilers? Normally uses of this are guarded by `#if defined(J9VM_OPT_JITSERVER)`.

Reply:
I had forgotten that not all of our build compilers supported it. That applies to the unordered maps as well, right? I can add a table myself.

Reply:
zOS is the most offending platform in terms of compiler support. Sometimes AIX gives us grief due to the xlC compiler. All the others should work.
When this PR is ready I am going to try to compile on all platforms (though zOS needs a different procedure)